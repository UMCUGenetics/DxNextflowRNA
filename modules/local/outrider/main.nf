process OUTRIDER {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "oras://community.wave.seqera.io/library/bioconductor-outrider_r-argparse_r-tidyverse:bf4b6d86b39e3c3e"


    input:
    tuple path(counts), path(refset)
    tuple val(meta), path(gtf)

    output:
    tuple val(meta), path("*.outrider_result.tsv"), emit: tsv
    tuple val(meta), path("counts_heatplots.pdf"), emit: psv

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def args = task.ext.args ?: ""
    def nthreads = task.cpus ?: 1
    def count_files = counts.join(" ")
    """
    outrider.R \\
        ${args} \\
        --ref ${refset} \\
        --pref ${prefix} \\
        --gtf ${gtf} \\
        --threads ${nthreads} \\
        ${count_files} \\
    """
}
