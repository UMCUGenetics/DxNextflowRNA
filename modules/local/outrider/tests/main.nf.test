nextflow_process {

    name "Test Process OUTRIDER"
    script "../main.nf"
    process "OUTRIDER"
    config "nextflow.config"

    tag "modules"
    tag "modules/local"
    tag "local"
    tag "outrider"
    tag "modules/outrider"


    test("Test outrider with simulated data") {
        when {
            process {
                """
                input[0] = Channel
                    .fromPath(params.outrider_test_data + "sim_strong/ref/*.tsv",
                    checkIfExists: true)
                    .toList()
                    .map { files ->
                        [
                [id: "test"],
                        file(params.outrider_test_data + "sim_strong/query/S01.tsv"),
                        files
                        ]
      }
                input[1] = [
                    [id: "gencodev44"],
                    file(params.outrider_test_data + "/sim_strong/sim.gtf")
                ]
                """
            }
        }
        then {

            assertAll (
                {assert process.success},
                {assert snapshot(process.out.tsv_signif).match()}
            )
        }
    }

}
