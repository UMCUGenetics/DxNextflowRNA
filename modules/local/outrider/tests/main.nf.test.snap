{
    "Test outrider with simulated data": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test_gene_outrider_result_signif.tsv:md5,af9253f88c1cdde0a9a826c4b03609fd"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-06T14:00:54.29771523"
    }
}